### PR TITLE
Add new design system colors / CSS variable clean up

### PR DESCRIPTION
**Pull request by maxxcrawford**
This PR does two things, but touches LOTS of files: 

- Adds the new Nebula design system colors to use while we figure out how to integrate into Protocol. 
- Migrate *most of the CSS variables to use Protocol tokens and the new Nebula color variables. 

Testing notes: 
- For the most part, the color variable updates remained the same value; just new tokens
- Using Protocol sizing tokens, so a few max/min widths adjusted by a -/+50 pixels. 

⚠️  Leaving the following alone for now: 
- Glocal / Bento box styles and variables
- `root:` custom gradients and drop shadow variables 

Other items: 

- [x] Add hex code lint rule
- [x] Add leading zero lint rule

### CODE REVIEW

**Comment by maxxcrawford** [comment]
@toufali @pdehaan I would love for you to review, but it's completely optional for ya. :)

**Comment by codemist** [comment]
The variable names for the Nebula colors look like they conflict with the ones for [Protocol](https://protocol.mozilla.org/fundamentals/color.html) which could be confusing. Some of the variables are same for both Nebula and Protocol though, like a few of the greens.

File: static/scss/includes/variables.scss [hunk 1]
```diff
@@ -6,3 +16,162 @@ $color-mpp-background: #E5E5E5;
...
+// Error
+$color-error: #FF4F5E;
+$color-error-active: #C50042;
+$color-error-hover: #E22850;
+$color-error-focus: #FFBDC5 ;
+
+// Warning
+$color-warning: #FFA436;
+$color-warning-active: #FFA436;
+$color-warning-hover: #FFA436;
```
Comment (pdehaan) [inline static/scss/includes/variables.scss:89]: Should the :hover have a different color?

File: static/scss/partials/dashboard.scss [hunk 2]
```diff
@@ -95,18 +95,11 @@
         box-shadow: var(--dropShadowLight);
```
Comment (pdehaan) [inline static/scss/partials/dashboard.scss:95]: Do we have strong feelings on the mix of `var()` syntax and using Sass variables?

File: static/scss/partials/dashboard.scss [hunk 3]
```diff
@@ -122,6 +115,12 @@
                 -o-transform:      rotate( 135deg );
                 -webkit-transform: rotate( 135deg );
```
Comment (pdehaan) [inline static/scss/partials/dashboard.scss:116]: Sorry, I haven't looked at this codebase in a long time, but do we want autoprefixer for adding vendor prefixes?

File: static/scss/partials/main.scss [hunk 4]
```diff
@@ -428,9 +388,9 @@ button:active {
 
 header {
   color: $color-white;
-  border-bottom: 1px solid var(--inkDark);
+  border-bottom: 1px solid rgba($color-grey-60, .5);
   background: var(--darkBlueGradient);
-  box-shadow: 0 0 3px 0 var(--inkDark);
+  box-shadow: 0 0 3px 0 rgba($color-grey-60, .5);
```
Comment (pdehaan) [inline static/scss/partials/main.scss:393]: nit: Not sure if we want to add a CSS linting rule re: prefixed zeros. I see we have `.5` here, but `0.5s` below on L404.

**Comment by maxxcrawford** [comment]
> The variable names for the Nebula colors look like they conflict with the ones for [Protocol](https://protocol.mozilla.org/fundamentals/color.html) which could be confusing. Some of the variables are same for both Nebula and Protocol though, like a few of the greens.

This was intentional for now. As Nebula is integrated into a theme of Protocol (current plan?) – we may have to update the name spaces to but I wanted to make adopting Protocol naming conventions as easy as possible out the gate.

File: static/scss/includes/variables.scss [hunk 5]
```diff
@@ -1,3 +1,13 @@
+// CSS Custom Properties
+:root {
+  --buttonFocus: 0 0 0 1px #0a84ff, 0 0 0 1px #0a84ff, 0 0 0 4px rgba(10, 132, 255, 0.3);
```
[the remark this answers is not in the document]
Comment (maxxcrawford) [reply static/scss/includes/variables.scss]: I was leaving the more intricate `--vars` alone for now. 

Good call about the hex codes. I added a rule, and auto-fix'd everything to uppercase.

File: static/scss/includes/variables.scss [hunk 6]
```diff
@@ -6,3 +16,162 @@ $color-mpp-background: #E5E5E5;
 $color-mpp-gray-30: #ECECEC;
 
 $box-shadow-mpp-sm: 0px 0px 4px rgba(0, 0, 0, 0.12);
```
[the remark this answers is not in the document]
Comment (maxxcrawford) [reply static/scss/includes/variables.scss]: Done! Added a no-leading-zero lint rule and fixed all the things.

File: static/scss/includes/variables.scss (hunk 1 above)
Comment (maxxcrawford) [reply static/scss/includes/variables.scss:89]: Great catch. There should be four separate warning colors. Fixing now!

File: static/scss/partials/dashboard.scss (hunk 2 above)
Comment (maxxcrawford) [reply static/scss/partials/dashboard.scss:95]: Yeah – Goal is to have no `--var/var()` variables.

File: static/scss/partials/dashboard.scss (hunk 3 above)
Comment (maxxcrawford) [reply static/scss/partials/dashboard.scss:116]: Nah – This smells like copy pasta. Fixing!

**Review by codemist — APPROVED** [review]

File: static/scss/includes/variables.scss [hunk 7]
```diff
@@ -1,3 +1,13 @@
+// CSS Custom Properties
+:root {
+  --buttonFocus: 0 0 0 1px #0a84ff, 0 0 0 1px #0a84ff, 0 0 0 4px rgba(10, 132, 255, 0.3);
+  --darkBlueGradient: linear-gradient(60deg, #2d264e, #26294e);
+  --dropShadowLight: 0 0 10px -3px #3d3f4740;
+  --dropShadowDark: 0 12px 18px 2px rgba(34, 0, 51, 0.09), 0 6px 22px 4px rgba(4, 25, 59, 0.87), 0 6px 10px -4px rgba(14, 13, 26, 0.12);
```
[the remark this answers is not in the document]
Comment (maxxcrawford) [reply static/scss/includes/variables.scss]: Leaving these as is for right now.

**Pull request merged**